File: python/10001st prime.py
from math import sqrt
import time

def is_prime(x):
    if x <= 1:
        return False
    else:
        for i in range(2, int(sqrt(x)) + 1):
            if x % i == 0:
                return False
        return True

primes = []
n = 1
while len(primes) < 10001:
    if is_prime(n):
        primes.append(n)
    n += 1

print(primes[-1])
print("time elapsed: ", time.process_time())

File: c/10001st prime.c
#include <stdlib.h>
#include <math.h>

int isPrime(int x)
{
    int i=5, stop = sqrt((double)x);
    if(x%2==0) return 0;
    if(x%3==0) return 0;
    while(i <= stop)
    {
        if(x%i==0 || x%(i+2)==0) return 0;
        i+=6;
    }
    return 1;
}

int main()
{
    int i=13, primeCount=5, lastPrime;
    while(primeCount<10001)
    {
        if(isPrime(i))
        {
            primeCount++;
            lastPrime = i;
        }
        i++;
    }
    return lastPrime;
}
